spring/src/main/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerFactoryProcessor.java

```diff
@@ -121,6 +61,75 @@ public int getOrder() {
   @Override
   public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry)
       throws BeansException {
-    this.beanDefinitionRegistry = registry;
+    if (registry instanceof BeanFactory) {
+      for (String name : registry.getBeanDefinitionNames()) {
+        Class<?> cls = ((BeanFactory) registry).getType(name);
+        if (cls != null) {
+          KubernetesInformers kubernetesInformers =
+              AnnotatedElementUtils.getMergedAnnotation(cls, KubernetesInformers.class);
+          if (kubernetesInformers != null && kubernetesInformers.value().length > 0) {
+            for (KubernetesInformer kubernetesInformer : kubernetesInformers.value()) {
+              registerInformer(registry, kubernetesInformer);
+              registerLister(registry, kubernetesInformer);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private void registerInformer(
+      BeanDefinitionRegistry registry, KubernetesInformer kubernetesInformer) {
+    RootBeanDefinition informerBean =
+        (RootBeanDefinition)
+            BeanDefinitionBuilder.rootBeanDefinition(SharedInformer.class).getBeanDefinition();
+    informerBean.setInstanceSupplier(
+        () -> informer(kubernetesInformer.apiTypeClass(), kubernetesInformer));
+    ResolvableType informerType =
+        ResolvableType.forClassWithGenerics(
+            SharedInformer.class, kubernetesInformer.apiTypeClass());
+    informerBean.setTargetType(informerType);
+    registry.registerBeanDefinition(kubernetesInformer.apiTypeClass().getName(), informerBean);
+  }
+
+  private void registerLister(
+      BeanDefinitionRegistry registry, KubernetesInformer kubernetesInformer) {
+    RootBeanDefinition listerBean =
+        (RootBeanDefinition)
```

Comment:
how about removing the cast ? or RootBeanDefinition  is the only way to call setTargetType?
how about using directly BeanDefinitionBuilder#genericBeanDefinition(java.lang.Class<T>, Supplier<T>) ? or same as above?

Reply:
It’s the only way. It’s safe by construction.